### PR TITLE
fix: image block migration

### PR DESCRIPTION
**Pull request by kellyjosephprice**
| [![PR App][icn]][demo] |
| :--------------------: |

## 🧰 Changes

Fixes migrating `image` magic blocks to `mdx`.

The processors failed to handle the following syntax correctly:

```
[block:image]
{ images: [{ ... }] }
[/block]
```

This adds a transformer for that case and reworks some of the transformers and compilers around images.

## 🧬 QA & Testing

Not totally sure how to test. Since the tests passed, I'm pretty happy?

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-PR_NUMBER.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io
[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

### CODE REVIEW

File: lib/mdx.ts [hunk 1]
```diff
@@ -15,8 +15,8 @@ export const mdx = (tree: any, { hast = false } = {}) => {
     .use(divTransformer)
     .use(readmeToMdx)
     .use(tablesToJsx)
-    .use(remarkStringify)
-    .use(compilers);
+    .use(compilers)
+    .use(remarkStringify);
```
Comment (kellyjosephprice) [inline lib/mdx.ts:19]: We're not doing it yet, but this changing the order allows us to override the defaults.

File: processor/compile/image.ts
Comment (kellyjosephprice) [inline processor/compile/image.ts:1]: I think it was a mistake to override the default `image` type from the start.

File: lib/mdx.ts (hunk 1 above)
Comment (rafegoldberg) [reply lib/mdx.ts:19]: ~Ahh, so is this how you were able to [delete the image compiler](https://github.com/readmeio/markdown/pull/951/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#diff-5e6825f007ffc0385b84a1770dd7496b4e7b9dea0fdd206860715060f1cc8ed8) wholesale?~ Or wait, it looks like we just [using the custom component *instead* of](https://github.com/readmeio/markdown/pull/951/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#diff-c626b24d42bb64ab7faddfb1f3a3e11f55b8dec3e952a1f13540ba7dd9621f13R48) that old custom compiler? This definitely feels more MDX-y, I think!

**Review by rafegoldberg — APPROVED** [review]
Yeah, Hard to test until we can pull it in alongside the pre-7.0 version in ReadMe proper, but I think this all makes sense to me (save for [one question](https://github.com/readmeio/markdown/pull/951/#discussion_r1729223847))! 🤘

File: lib/mdx.ts (hunk 1 above)
Comment (kellyjosephprice) [reply lib/mdx.ts:19]: Yea, this PR detects earlier (with a transformer) if it needs MDX syntax. That way we're relying on the default compilers for both MDX images and markdown images.

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v7.2.1</b>](/readmeio/markdown/releases/tag/v7.2.1)
<br>

**Pull request merged**